### PR TITLE
Socint 412

### PR DESCRIPTION
**Pull request by KieranWardle**
https://collaborate2.ons.gov.uk/jira/browse/SOCINT-412
Added audits for changes to users, roles and permissions 

Tested in cc-dev

### CODE REVIEW

**Review by PeterBochel — APPROVED** [review]
Approved, with 1 minor comment

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/endpoint/UserEndpoint.java [hunk 1]
```diff
@@ -46,9 +53,15 @@
...
-  public UserEndpoint(final RBACService rbacService, final UserService userService) {
+  public UserEndpoint(
+      final RBACService rbacService,
+      final UserService userService,
+      final UserAuditService userAuditService,
+      final RoleService roleService) {
     this.rbacService = rbacService;
     this.userService = userService;
+    this.userAuditService = userAuditService;
+    this.roleService = roleService;
```
[the remark this answers is not in the document]
Comment (KieranWardle) [reply src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/endpoint/UserEndpoint.java]: Oops no it's not, removed

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/model/AuditType.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+public enum AuditType {
+  USER(new AuditTarget[] {AuditTarget.USER}, CREATED, MODIFIED, REMOVED),
+  USER_SURVEY_USAGE(new AuditTarget[] {AuditTarget.USER}, ADDED, REMOVED),
+  USER_ROLE(new AuditTarget[] {AuditTarget.USER, AuditTarget.ROLE}, ADDED, REMOVED),
+  ADMIN_ROLE(new AuditTarget[] {AuditTarget.USER, AuditTarget.ROLE}, ADDED, REMOVED),
+  ROLE(new AuditTarget[] {AuditTarget.ROLE}, CREATED),
+  PERMISSION(new AuditTarget[] {AuditTarget.ROLE}, ADDED, REMOVED),
+  LOGIN(new AuditTarget[] {AuditTarget.ROLE}),
+  LOGOUT(new AuditTarget[] {AuditTarget.ROLE});
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/model/AuditType.java:21]: The User is the target for login/out, not the role. WHO logged in/out

**Review by philwhiles — CHANGES_REQUESTED** [review]
see comment in audittype

**Pull request merged**